expo-navigation-bar methods creates Java.lang.NoSuchMethodError and java.lang.NoSuchMethodError

### Summary

App crashes on startup in devices with sdk 29, 28 and 27 
here is a complete description of the errors.

```
FATAL EXCEPTION: main
Process: com.vdiorio.EdhCounter, PID: 22481
java.lang.NoSuchMethodError: No static method navigationBars()I in class Landroid/view/WindowInsets$Type; or its super classes (declaration of 'android.view.WindowInsets$Type' appears in /system/framework/framework.jar!classes3.dex)
	at expo.modules.navigationbar.NavigationBarModule$getVisibilityAsync$1.invoke(NavigationBarModule.kt:115)
	at expo.modules.navigationbar.NavigationBarModule$getVisibilityAsync$1.invoke(NavigationBarModule.kt:20)
	at expo.modules.navigationbar.NavigationBarModule$safeRunOnUiThread$1.run(NavigationBarModule.kt:43)
	at android.os.Handler.handleCallback(Handler.java:883)
	at android.os.Handler.dispatchMessage(Handler.java:100)
	at android.os.Looper.loop(Looper.java:237)
	at android.app.ActivityThread.main(ActivityThread.java:8016)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1076)
```

```
FATAL EXCEPTION: main
Process: com.vdiorio.EdhCounter, PID: 11714
java.lang.NoClassDefFoundError: Failed resolution of: Landroid/view/WindowInsets$Type;
	at expo.modules.navigationbar.NavigationBarModule$getVisibilityAsync$1.invoke(NavigationBarModule.kt:115)
	at expo.modules.navigationbar.NavigationBarModule$getVisibilityAsync$1.invoke(NavigationBarModule.kt:20)
	at expo.modules.navigationbar.NavigationBarModule$safeRunOnUiThread$1.run(NavigationBarModule.kt:43)
	at android.os.Handler.handleCallback(Handler.java:873)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:193)
	at android.app.ActivityThread.main(ActivityThread.java:6718)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
Caused by: java.lang.ClassNotFoundException: Didn't find class "android.view.WindowInsets$Type" on path: DexPathList[[zip file "/data/app/com.vdiorio.EdhCounter-OvWNu2PrYwn0MPLAiB5o1A==/base.apk", zip file "/data/app/com.vdiorio.EdhCounter-OvWNu2PrYwn0MPLAiB5o1A==/split_config.arm64_v8a.apk", zip file "/data/app/com.vdiorio.EdhCounter-OvWNu2PrYwn0MPLAiB5o1A==/split_config.xxhdpi.apk"],nativeLibraryDirectories=[/data/app/com.vdiorio.EdhCounter-OvWNu2PrYwn0MPLAiB5o1A==/lib/arm64, /data/app/com.vdiorio.EdhCounter-OvWNu2PrYwn0MPLAiB5o1A==/base.apk!/lib/arm64-v8a, /data/app/com.vdiorio.EdhCounter-OvWNu2PrYwn0MPLAiB5o1A==/split_config.arm64_v8a.apk!/lib/arm64-v8a, /data/app/com.vdiorio.EdhCounter-OvWNu2PrYwn0MPLAiB5o1A==/split_config.xxhdpi.apk!/lib/arm64-v8a, /system/lib64]]
	at dalvik.system.BaseDexClassLoader.findClass(BaseDexClassLoader.java:134)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:379)
	at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
	... 10 more
	Suppressed: java.io.IOException: No original dex files found for dex location /data/app/com.vdiorio.EdhCounter-OvWNu2PrYwn0MPLAiB5o1A==/split_config.arm64_v8a.apk
		at dalvik.system.DexFile.openDexFileNative(Native Method)
		at dalvik.system.DexFile.openDexFile(DexFile.java:354)
		at dalvik.system.DexFile.<init>(DexFile.java:101)
		at dalvik.system.DexFile.<init>(DexFile.java:75)
		at dalvik.system.DexPathList.loadDexFile(DexPathList.java:394)
		at dalvik.system.DexPathList.makeDexElements(DexPathList.java:354)
		at dalvik.system.DexPathList.<init>(DexPathList.java:164)
		at dalvik.system.BaseDexClassLoader.<init>(BaseDexClassLoader.java:74)
		at dalvik.system.BaseDexClassLoader.<init>(BaseDexClassLoader.java:65)
		at dalvik.system.PathClassLoader.<init>(PathClassLoader.java:64)
		at com.android.internal.os.ClassLoaderFactory.createClassLoader(ClassLoaderFactory.java:73)
		at com.android.internal.os.ClassLoaderFactory.createClassLoader(ClassLoaderFactory.java:88)
		at android.app.ApplicationLoaders.getClassLoader(ApplicationLoaders.java:74)
		at android.app.ApplicationLoaders.getClassLoader(ApplicationLoaders.java:40)
		at android.app.LoadedApk.createOrUpdateClassLoaderLocked(LoadedApk.java:727)
		at android.app.LoadedApk.getClassLoader(LoadedApk.java:810)
		at android.app.LoadedApk.getResources(LoadedApk.java:1032)
		at android.app.ContextImpl.createAppContext(ContextImpl.java:2345)
		at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5798)
		at android.app.ActivityThread.access$1100(ActivityThread.java:200)
		at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656)
		at android.os.Handler.dispatchMessage(Handler.java:106)
		... 5 more
	Suppressed: java.io.IOException: No original dex files found for dex location /data/app/com.vdiorio.EdhCounter-OvWNu2PrYwn0MPLAiB5o1A==/split_config.xxhdpi.apk
		at dalvik.system.DexFile.openDexFileNative(Native Method)
		at dalvik.system.DexFile.openDexFile(DexFile.java:354)
		at dalvik.system.DexFile.<init>(DexFile.java:101)
		at dalvik.system.DexFile.<init>(DexFile.java:75)
		at dalvik.system.DexPathList.loadDexFile(DexPathList.java:394)
		at dalvik.system.DexPathList.makeDexElements(DexPathList.java:354)
		at dalvik.system.DexPathList.<init>(DexPathList.java:164)
		at dalvik.system.BaseDexClassLoader.<init>(BaseDexClassLoader.java:74)
		at dalvik.system.BaseDexClassLoader.<init>(BaseDexClassLoader.java:65)
		at dalvik.system.PathClassLoader.<init>(PathClassLoader.java:64)
		at com.android.internal.os.ClassLoaderFactory.createClassLoader(ClassLoaderFactory.java:73)
		at com.android.internal.os.ClassLoaderFactory.createClassLoader(ClassLoaderFactory.java:88)
		at android.app.ApplicationLoaders.getClassLoader(ApplicationLoaders.java:74)
		at android.app.ApplicationLoaders.getClassLoader(ApplicationLoaders.java:40)
		at android.app.LoadedApk.createOrUpdateClassLoaderLocked(LoadedApk.java:727)
		at android.app.LoadedApk.getClassLoader(LoadedApk.java:810)
		at android.app.LoadedApk.getResources(LoadedApk.java:1032)
		at android.app.ContextImpl.createAppContext(ContextImpl.java:2345)
		at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5798)
		at android.app.ActivityThread.access$1100(ActivityThread.java:200)
		at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1656)
		at android.os.Handler.dispatchMessage(Handler.java:106)
		... 5 more
```

### Managed or bare workflow? If you have `ios/` or `android/` directories in your project, the answer is bare!

managed

### What platform(s) does this occur on?

Android

### SDK Version (managed workflow only)

44

### Environment

  expo-env-info 1.0.2 environment info:
    System:
      OS: Linux 5.10 Ubuntu 20.04.3 LTS (Focal Fossa)
      Shell: 5.8 - /usr/bin/zsh
    Binaries:
      Node: 14.19.0 - ~/.nvm/versions/node/v14.19.0/bin/node
      Yarn: 1.22.17 - /usr/local/bin/yarn
      npm: 6.14.16 - ~/.nvm/versions/node/v14.19.0/bin/npm
    npmPackages:
      expo: 43.0.3 => 44.0.6 
      react: 17.0.1 => 17.0.1 
      react-dom: 17.0.1 => 17.0.1 
      react-native: 0.64.3 => 0.64.3 
      react-native-web: 0.17.1 => 0.17.1 
    npmGlobalPackages:
      eas-cli: 0.47.0
      expo-cli: 5.2.0
    Expo Workflow: managed

### Reproducible demo

Simply use any method of expo-navigation-bar in application and build it

--- a/packages/expo-navigation-bar/CHANGELOG.md
+++ b/packages/expo-navigation-bar/CHANGELOG.md
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- Fix `getVisiblilityAsync` crashing on Android 10 and older.
+
 ### 💡 Others
 
 - Updated `@expo/config-plugins` from `4.0.2` to `4.0.14` ([#15621](https://github.com/expo/expo/pull/15621) by [@EvanBacon](https://github.com/EvanBacon))

--- a/packages/expo-navigation-bar/CHANGELOG.md
+++ b/packages/expo-navigation-bar/CHANGELOG.md
@@ -8,7 +8,7 @@
 
 ### 🐛 Bug fixes
 
-- Fix `getVisiblilityAsync` crashing on Android 10 and older.
+- Fix `getVisiblilityAsync` crashing on Android 10 and older. ([#16445](https://github.com/expo/expo/pull/16445) by [@barthap](https://github.com/barthap))
 
 ### 💡 Others
 

--- a/packages/expo-navigation-bar/android/src/main/java/expo/modules/navigationbar/NavigationBarModule.kt
+++ b/packages/expo-navigation-bar/android/src/main/java/expo/modules/navigationbar/NavigationBarModule.kt
@@ -111,14 +111,14 @@ class NavigationBarModule(context: Context) : ExportedModule(context) {
   @ExpoMethod
   fun getVisibilityAsync(promise: Promise) {
     safeRunOnUiThread(promise) {
-      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-        val visibility = if (it.window.decorView.rootWindowInsets.isVisible(WindowInsets.Type.navigationBars())) "visible" else "hidden"
-        promise.resolve(visibility)
+      val isVisible = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+        it.window.decorView.rootWindowInsets.isVisible(WindowInsets.Type.navigationBars())
       } else {
-        // TODO: Verify this works
-        @Suppress("DEPRECATION") val visibility = if ((View.SYSTEM_UI_FLAG_HIDE_NAVIGATION and it.window.decorView.systemUiVisibility) == 0) "visible" else "hidden"
-        promise.resolve(visibility)
+        @Suppress("DEPRECATION")
+        (View.SYSTEM_UI_FLAG_HIDE_NAVIGATION and it.window.decorView.systemUiVisibility) == 0
       }
+      val visibility = if (isVisible) "visible" else "hidden"
+      promise.resolve(visibility)
     }
   }
 

